File: VB/VB/WinWebSolution.Module.Win/WinUIContainerViewItem.vb
Imports Microsoft.VisualBasic
Imports System
Imports WinCustomUserControlLibrary
Imports DevExpress.ExpressApp.Editors
Imports DevExpress.ExpressApp.Model

Namespace WinWebSolution.Module.Win
	Public Interface IWinUIContainerViewItem
	Inherits IModelViewItem
	End Interface
	<DetailViewItemAttribute(GetType(IWinUIContainerViewItem))> _
	Public Class WinUIContainerViewItem
		Inherits ViewItem
		Public Sub New(ByVal objectType As Type, ByVal id As String)
			MyBase.New(objectType, id)
		End Sub
		Public Sub New(ByVal model As IWinUIContainerViewItem, ByVal objectType As Type)
			MyBase.New(objectType, model.Id)
		End Sub
		Protected Overrides Function CreateControlCore() As Object
			Return New WinCustomUserControl()
		End Function
	End Class
End Namespace

File: CS/WinWebSolution.Module.Win/WinUIContainerViewItem.cs
using System;
using WinCustomUserControlLibrary;
using DevExpress.ExpressApp.Editors;
using DevExpress.ExpressApp.Model;

namespace WinWebSolution.Module.Win {
    public interface IWinUIContainerViewItem: IModelViewItem { }
    [DetailViewItemAttribute(typeof(IWinUIContainerViewItem))]
    public class WinUIContainerViewItem : ViewItem {
        public WinUIContainerViewItem(Type objectType, string id) : base(objectType, id) { }
        public WinUIContainerViewItem(IWinUIContainerViewItem model, Type objectType) : base(objectType, model.Id) { }
        protected override object CreateControlCore() { 
            return new WinCustomUserControl(); 
        }
    }
}